### PR TITLE
Make pass renderer configurable & other fixes

### PR DESCRIPTION
**Pull request by dmach**
### What does this PR do?
The pass renderer becomes configurable.
Also several issues in the code have been fixed.

### Previous Behavior
No changes to the existing behavior, new features must be explicitly enabled.

### New Behavior
Config option `pass_variable_prefix` allows to distinguish variables that contain paths to pass secrets.
Config option `pass_strict_fetch` allows to error out when a secret cannot be fetched from pass.
Config option ` pass_dir` allows setting the `PASSWORD_STORE_DIR` env for pass.
Config option ` pass_gnupghome` allows setting the `$GNUPGHOME` env for pass.
Pass executable path from _get_path_exec() is used when calling the program.
The `$HOME` env is no longer modified globally.
Only trailing newlines are stripped from the fetched secret.
Pass process arguments are handled in a secure way.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by meaksh — COMMENTED** [review]
Thanks for this PR! The code looks good to me :+1:

It would be definetely great that you could also provide some tests here.

**Review by meaksh — DISMISSED** [review]
Awesome! Thanks for the new enhancements and unit tests! :+1:

LGTM

(BTW: the failing test does look related to the changes in this PR)

**Comment by dmach** [comment]
> (BTW: the failing test does look related to the changes in this PR)

You mean this one? https://jenkins.saltproject.io/job/pr-macosx-catalina-x86_64-py3-pytest/job/PR-62120/6/
That failed in the `tests.unit.utils.test_verify.TestVerify.test_max_open_files` test - which is most likely unrelated.

**Comment by meaksh** [comment]
@dmach sorry I meant it does **not** look related :sweat_smile: . Thanks!

**Comment by dmach** [comment]
@meaksh @krionbsd Is there something I can do to move this forward?

**Comment by meaksh** [comment]
@Ch3LL any chance to have a review of this one? Thanks in advance!

File: salt/renderers/pass.py [hunk 1]
```diff
@@ -75,18 +99,66 @@ def _fetch_secret(pass_path):
...
+    pass_exec = _get_pass_exec()
+
+    # Make a backup in case we want to return the original value without stripped whitespaces
+    original_pass_path = pass_path
+
+    # Remove the optional prefix from pass path
+    pass_prefix = __opts__.get("pass_variable_prefix", None)
+    if pass_prefix:
+        # If we do not see our prefix we do not want to process this variable
+        if not pass_path.startswith(pass_prefix):
```
Comment (Ch3LL) [inline salt/renderers/pass.py:111]: Can you make sure we document for users that if the the pass path doesn't start with the prefix the renderer will return the path.

File: salt/renderers/pass.py [hunk 2]
```diff
@@ -45,6 +45,30 @@
 
         pass:
           pkg.installed
+
+Configuration options
```
Comment (Ch3LL) [inline salt/renderers/pass.py:49]: Can you clarify for the user which config file these config options need to be set.

File: salt/renderers/pass.py [hunk 3]
```diff
@@ -45,6 +45,30 @@
...
+
+.. code-block:: yaml
+
+        # If the prefix is set, only the prefixed template variables are
+        # considered for fetching secrets from Pass. The default behavior
+        # is that *all* variables are considered.
+        # When the prefix matching is on, You may also consider
+        # setting the `renderer: jinja|yaml|pass` config option.
+        # recommended value: 'pass:'
+        pass_variable_prefix: 'pass:'
```
Comment (Ch3LL) [inline salt/renderers/pass.py:59]: can you make sure all of these new options are also added to salt/config/__init__.py with their defaults set.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/renderers/pass.py (hunk 3 above)
Comment (dmach) [reply salt/renderers/pass.py:64]: Done

File: salt/renderers/pass.py (hunk 1 above)
Comment (dmach) [reply salt/renderers/pass.py:116]: I've rephrased the documentation to make this more clear.

File: salt/renderers/pass.py (hunk 2 above)
Comment (dmach) [reply salt/renderers/pass.py]: I've changed `Configuration options` to `Salt master configuration options`

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
lint and pre-commit are failing. Once we get that fixed up we can get this one merged in.

**Comment by dmach** [comment]
> lint and pre-commit are failing. Once you get that fixed up we can get this one merged in.

Fixed (I hope). Other 2 tests are failing now, but I don't think they're related.

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
For some reason I cannot update the branch. Can you rebase and push? We shouldn't be seeing those test issues so lets re-base and start them again. thanks

**Comment by dmach** [comment]
> For some reason I cannot update the branch. Can you rebase and push? We shouldn't be seeing those test issues so lets re-base and start them again. thanks

Done via GitHub webui.
I was also quite surprised seeing the test issues. Looks like a caching problem to me - jenkins did not know about the commit I just pushed.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**